src/observerClass.js

```diff
@@ -35,7 +35,7 @@ export function makeClassComponentObserver(componentClass) {
     }
     patch(target, "componentWillUnmount", function() {
         if (isUsingStaticRendering() === true) return
-        this.render[mobxAdminProperty] && this.render[mobxAdminProperty].dispose()
+        this[mobxAdminKeyForRender] && this[mobxAdminKeyForRender].dispose()
```

Comment:
What if we actually keep in `render` as well for compatibility purposes, but disposal will happen based on what value from the instance?